### PR TITLE
fix: IDP login button styles

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: migrations/cockroach/V1.19__idp_configs.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+ALTER TABLE management.idp_configs DROP COLUMN logo_src;
+ALTER TABLE adminapi.idp_configs DROP COLUMN logo_src;
+ALTER TABLE auth.idp_configs DROP COLUMN logo_src;
```
Comment (livio-a) [inline migrations/cockroach/V1.19__idp_configs.sql:3]: i believe these will fail as sql_safe_update is on...
@adlerhurst does it work with flyway?

File: internal/ui/login/static/templates/register_option.html [hunk 2]
```diff
@@ -19,7 +19,7 @@ <h1>{{t "RegisterOption.Title"}}</h1>
         {{if .LoginPolicy.AllowExternalIDP}}
             {{ $reqid := .AuthReqID}}
             {{range $provider := .IDPProviders}}
-                <a href="{{ externalIDPRegisterURL $reqid $provider.IDPConfigID}}" class="button secondary idp-providers">
+                <a href="{{ externalIDPRegisterURL $reqid $provider.IDPConfigID}}" class="button secondary idp-providers {{idpProviderClass $provider.StylingType}}">
                     {{$provider.Name}}
```
Comment (livio-a) [inline internal/ui/login/static/templates/register_option.html:23]: don't we want / have to display the logo here as well?

File: internal/ui/login/static/templates/login.html [hunk 3]
```diff
@@ -42,8 +42,9 @@ <h1>{{t "Login.Title"}}</h1>
 
             {{ $reqid := .AuthReqID}}
             {{range $provider := .IDPProviders}}
-                <a href="{{ externalIDPAuthURL $reqid $provider.IDPConfigID}}" class="button secondary idp-providers">
-                    {{$provider.Name}}
+                <a href="{{ externalIDPAuthURL $reqid $provider.IDPConfigID}}" class="button secondary idp-provider {{idpProviderClass $provider.StylingType}}">
+                    <span class="logo"></span>
+                    <span class="provider-name">{{$provider.Name}}</span>
```
Comment (livio-a) [inline internal/ui/login/static/templates/login.html:47]: what happens if stylingType is empty? did you try as well?

File: internal/ui/login/handler/register_option_handler.go [hunk 4]
```diff
@@ -35,7 +36,15 @@ func (l *Login) renderRegisterOption(w http.ResponseWriter, r *http.Request, aut
...
 		baseData: l.getBaseData(r, authReq, "RegisterOption", errType, errMessage),
 	}
-	l.renderer.RenderTemplate(w, r, l.renderer.Templates[tmplRegisterOption], data, nil)
+	funcs := map[string]interface{}{
+		"hasExternalLogin": func() bool {
+			return authReq.LoginPolicy.AllowExternalIDP && authReq.AllowedExternalIDPs != nil && len(authReq.AllowedExternalIDPs) > 0
+		},
+		"idpProviderClass": func(stylingType iam_model.IDPStylingType) string {
+			return stylingType.GetCSSClass()
+		},
```
Comment (livio-a) [inline internal/ui/login/handler/register_option_handler.go:45]: not really necessary... the function is already implemented in renderer.go

File: internal/ui/login/handler/login_handler.go [hunk 5]
```diff
@@ -72,6 +73,9 @@ func (l *Login) renderLogin(w http.ResponseWriter, r *http.Request, authReq *mod
 		"hasExternalLogin": func() bool {
 			return authReq.LoginPolicy.AllowExternalIDP && authReq.AllowedExternalIDPs != nil && len(authReq.AllowedExternalIDPs) > 0
 		},
+		"idpProviderClass": func(stylingType iam_model.IDPStylingType) string {
+			return stylingType.GetCSSClass()
+		},
```
Comment (livio-a) [inline internal/ui/login/handler/login_handler.go:78]: not really necessary... the function is already implemented in renderer.go

File: internal/ui/login/static/resources/themes/scss/variables.scss [hunk 6]
```diff
@@ -33,11 +33,15 @@ $inputBackgroundColor: #252525;
 $buttonBackgroundColorHover: $inputBackgroundColor;
 $profileImgDark: "../../../images/icon-user-dark";
 $otherUserImgDark: "../../../images/icon-newuser-dark";
+$idpGoogleImg: "../../../images/idp/google2";
```
Comment (livio-a) [inline internal/ui/login/static/resources/themes/scss/variables.scss:36]: there are 3 google logos uploaded in this PR but only 1 is referenced here

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/ui/login/static/templates/login.html (hunk 3 above)
Comment (fgerschwiler) [reply internal/ui/login/static/templates/login.html:47]: yes, than its styled as before

File: internal/ui/login/static/resources/themes/scss/variables.scss (hunk 6 above)
Comment (fgerschwiler) [reply internal/ui/login/static/resources/themes/scss/variables.scss:36]: removed 2 of them

**Review by livio-a — APPROVED** [review]

**Pull request merged**